### PR TITLE
feat: #2136 Trigger relogin with deep backlink to self, if user does …

### PR DESCRIPTION
**Pull request by markkopaas**
…not have permissions for the selected resource

### CODE REVIEW

File: src/pages/concept-list-page/concept-list-page.ts [hunk 1]
```diff
@@ -1,13 +1,17 @@
+import React, { ComponentType } from 'react';
```
Comment (azaxarov) [inline src/pages/concept-list-page/concept-list-page.ts]: Unused import: ComponentType

File: src/pages/concept-list-page/concept-list-page.ts [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 import _ from 'lodash';
 import { compose, withProps } from 'recompose';
 
 import { ConceptListPagePure } from './concept-list-page-pure';
 import { conceptListResolver } from './concept-list-resolver';
+import { hasPermission, logout } from '../../auth/auth-service';
```
Comment (azaxarov) [inline src/pages/concept-list-page/concept-list-page.ts]: Unused imports: hasPermission, logout

File: src/auth/auth-service.ts [hunk 3]
```diff
@@ -36,3 +37,14 @@ export const getToken = async () => {
...
+//expand authorities protocol to object structure
+const extractResourceRoles = (authorities: string) =>
+  authorities
+    .split(',')
+    .map(authorityDescriptor => authorityDescriptor.split(':'))
+    .map(parts => ({ resource: parts[0], resourceId: parts[1], role: parts[2] }));
+
+//check if user has any role in selected resource
+export const hasPermission: (string, resourceId) => boolean = (resource, resourceId) =>
+  !!find(extractResourceRoles(get(kc.tokenParsed, 'authorities')), { resource, resourceId });
```
Comment (azaxarov) [inline src/auth/auth-service.ts]: Is it possible to make this easier to follow? Not very clear as to what is going on here.

Comment (markkopaas) [reply src/auth/auth-service.ts]: det var før ;) 
noen ganger lønner dert ikke seg å fjerne navner og ha enlinjer

File: src/pages/concept-registration-page/concept-registration-resolver.ts [hunk 4]
```diff
@@ -6,4 +6,4 @@ const mapProps = {
     (await getConcept(`?orgNummer=${catalogId}`)).find(concept => concept.id === conceptId)
 };
 
-export const ConceptRegistrationResolver = resolve(mapProps);
+export const conceptRegistrationResolver = resolve(mapProps);
```
Comment (azaxarov) [inline src/pages/concept-registration-page/concept-registration-resolver.ts]: Unintentional replace?

**Pull request merged**